### ISSUE TITLE
CI всему голова

### ISSUE THREAD

**Issue by afaikiac** [rominf/latex-g7-32#131]
Доделал нормальный CI в своем репозитории: https://github.com/afaikiac/latex-g7-32

- Триггерит на коммиты в дефолтную ветку и PR.
- Если ветка дефолтная, то создается _драфт_ релиз с тегом в виде таймштампа. Мне такой подход показался более удобным, нежели вручную проставлять теги.

Пример PR с ошибкой который не собрался: https://github.com/afaikiac/latex-g7-32/pull/4

Если нравится, то:

1. Можно создать новый репозиторий с Init commit :) например, latex-g7-32-overleaf
2. Могу портировать старый шаблон, либо можно оставить мой.
3. По структуре есть замечания?
4. Можете добавить меня в организацию и я все настрою, как у себя. Можно и без этого, тогда расскажу как сделать, если надо.

**Comment by afaikiac** [comment]
Потом, при желании, можно и любую систему сборки сделать по образу и подобию в отдельном репозитории и со своим package.

**Comment by rominf** [comment]
Спасибо! У @einhander есть права, чтобы добавить вас, но если что - пишите, добавлю. Сам пока добавлять не буду, поскольку @einhander уже давно владеет проектом больше, чем я.

Прикреплю https://github.com/latex-g7-32/latex-g7-32/issues/43 для лучшей связности.

**Comment by rominf** [comment]
@afaikiac 

> Можно создать новый репозиторий с Init commit :)

Вместо основного, чтобы почистить историю?

Красивая история, это, конечно, хорошо, но я бы не стал:
1. Теряются звёзды, которые столько накапливались. Это точно скажется на популярности проекта.
2. Теряется вклад авторов. Вообще не уверен, что так можно делать, не нарушает ли это права авторов.

Я думаю, что мало кто смотрит на историю в git, скорее смотрят на README и незакрытые проблемы (см. #133).

Поправьте меня, пожалуйста, если я неправильно понял предложение.

**UPD0:**

Возможно, если сделать это аккуратно, то может быть, это будет и хорошо:

1. Сделать новый репозиторий:
1.1. Включить обязательную подпись коммитов (https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).
1.2. `Init commit`: добавить информацию о всех авторах (взять из истории + README) через co-author (https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).
1.3. Добавить файл AUTHORS (https://opensource.google/documentation/reference/releasing/authors/).
2. Старый (этот) репозиторий оставить, но:
2.1. Переименовать README в README.old.
2.2. Новый README сделать пустым со ссылкой на новый репозиторий. Также добавить ссылку в описание.
2.3. Выполнить #133, оставшиеся проблемы перенести в новый репозиторий.
2.4. Репозиторий заархивировать.

**UPD1:**
Будет 3 репозитория: `latex-g7-32` (заархивированный), `style` и `template`.

**Comment by afaikiac** [comment]
> Поправьте меня, пожалуйста, если я неправильно понял предложение.

Ни в коем случае не предлагаю удалять оригинальный репозиторий, там много интересных наработок. Предлагаю создать новый.

Я это вижу так:

- Новый репозиторий latex-g7-32-overleaf

Будет максимально простая ситема сборки с latexmk, которая должна в первую очередь совместима с overleaf.

- Оригинальный репозиторий

Для энтузиастов. Тоже с CI и со всеми системами сборки. Они все хороши и смысла их удалять нет.

---

Почему лучше новый:

- дает 3 варианта, как собрать документы

- к нему будет прикреплен образ, который весит всего 900M. К слову, я успешно его использую в однострочном скрипте в связке c vimtex.

- не будет ничего лишнего и он будет, как github template

- из-за особенностей оригинального репозитория c разнообразными (файл -> картинка), я не понимаю, как красиво его портировать в оверлиф со всеми фишечками моего форка.

Для репозитория latex-g7-32 образ уже весит 1.5G. По-хорошему, нужно заменить inkscape, и он станет поменьше. И чтобы настроить в нем нормальный CI, надо допиливать все скрипты, продумать, как лучше это обернуть в Docker, и полностью переписать README.

**Comment by afaikiac** [comment]
> Будет 3 репозитория: latex-g7-32 (заархивированный), style и template.

Да, template звучит хорошо, но мы выносили G7-32 чтобы постепенно доработать его до CTAN пакета. Не лучше ли будет оставить название репозитория для стиля, как есть?

> 1.3. Добавить файл AUTHOR

Мне кажется такой файл в шаблоне будет избыточен. Тем более, что это будет просто пример текста c настроенным CI :)

С остальным полностью солидарен.

**Comment by rominf** [comment]
Поскольку я не только давно отошёл от учебно-научной деятельности, но и даже сменил страну проживания на несовместимую с этим ГОСТом, будущее этого проекта не в моих руках, а в руках тех, кому он нужен.

Я испугался, что будут слишком радикальные изменения, которые могут сильно уменьшить популярность проекта, но оказалось, что я неправильно понял предложение; всё хорошо.

Думаю, что было бы хорошо не просто добавить вас в организацию, но и дать вам права владельца, поскольку вы сейчас активно участвуете в развитии проекта. Это также позволит несколько улучить bus factor (уже сталкивались с этой проблемой из-за моего недосмотра: https://github.com/latex-g7-32/latex-g7-32/issues/103#issuecomment-527032502). Но окончательное решение пусть примет @einhander.

**Comment by afaikiac** [comment]
Моей целью изначально было привнести сюда наработки, чтобы порог входа в "диплом на техе" был ниже, а также сделать почву для разработки CTAN пакета. Этот стиль очень хорош, но требует полного рефакторинга. Я не хочу становиться мейнтейнером,  а хочу причесать репозиторий для следующего энтузиаста.

**Comment by einhander** [comment]
@afaikiac если сможете настроить CI будет здорово. У меня времени катастрофически не хватает, чтобы сделать что-то полезное.

**Comment by afaikiac** [comment]
@einhander, добавьте меня в организацию, и я все сделаю. А именно:
1. создам репозиторий под названием `template`
2. создам package
3. сделаю порт текста из оригинального latex-g7-32 и добавлю CI

Первый коммит будет от имени людей, кто участвовал в создании текста шаблона.
Лицензию добавлять не буду, тк это просто шаблон. Как закончу, выйду.

**Comment by einhander** [comment]
@afaikiac Добавил
